Cms/ProfileBioEdit.ascx.cs

```diff
@@ -0,0 +1,2394 @@
+                    {
+                        ID = "lbAddFamilyMember",
+                        CssClass = "btn btn-primary btn-sm mt-2 pull-right",
+                        CausesValidation = false
+                    };
+                    lbAddFamilyMember.Click += lbAddFamilyMember_Click;
+                    lbAddFamilyMember.Controls.Add( new LiteralControl( "<i class='fa fa-plus'></i>" ) );
+                    lbAddFamilyMember.Visible = GetAttributeValue( AttributeKey.AllowAddingFamilyMembers ).AsBoolean();
+                    if ( pnlFamilyMemberHeader != null )
+                    {
+                        pnlFamilyMemberHeader.Controls.AddAt( 0, lbAddFamilyMember );
+                    }
+                    else
+                    {
+                        pnlFamilyMemberBody.Controls.AddAt( 0, lbAddFamilyMember );
+                    }
+
+                    var pnlFamilyMemberWidgets = new Panel { ID = "pnlFamilyMemberWidgets" };
+                    pnlFamilyMemberBody.Controls.Add( pnlFamilyMemberWidgets );
+
+                    foreach ( var fm in FamilyMembers )
+                    {
+                        AddFamilyMember( pnlFamilyMemberWidgets, fm );
+                    }
+                }
+            }
+            #endregion
+
+        }
+
+        private void AddFamilyMember( Control pnlFamilyMemberBody, PersonFamilyMember fm, bool expanded = false )
+        {
+            var pwFamilyMember = new PanelWidget
+            {
+                ID = $"pwFamilyMember_{fm.PersonId}",
+                Title = $"<h6>{fm.Person.FullName}<br><small>{fm.Person.GetFamilyRole()}</small></h6>",
+                CssClass = "family-member",
+                Expanded = expanded,
+                ShowDeleteButton = fm.Guid == Guid.Empty
+            };
+            pwFamilyMember.DeleteClick += pwFamilyMember_DeleteClick;
+
+            var pnlFamilyMemberPerson = new Panel { ID = "pnlFamilyMemberPerson", CssClass = "d-flex flex-wrap" };
+            pwFamilyMember.Controls.Add( pnlFamilyMemberPerson );
+
+            pnlFamilyMemberBody.Controls.Add( pwFamilyMember );
+
+            GeneratePersonFields( fm.Guid, pnlFamilyMemberPerson, null, true );
+            GenerateContactFields( fm.Guid, pnlFamilyMemberPerson, true );
+
+            if ( expanded )
+            {
+                pwFamilyMember.Focus();
+                ScriptManager.RegisterStartupScript( pwFamilyMember, pwFamilyMember.GetType(), $"focus-{pwFamilyMember.ClientID}", $"$('html, body').animate({{scrollTop: $('#{pwFamilyMember.ClientID}').offset().top }}, 1000);", true );
+            }
+        }
+
+        private void GeneratePersonFields( Guid personGuid, Panel pnlPerson, Control pnlFamilyBody, bool familyMember = false )
+        {
+            WebControl lSpacer = new WebControl( HtmlTextWriterTag.Span ) { ID = "PersonSpacer" };
+            var childGuid = Rock.SystemGuid.GroupRole.GROUPROLE_FAMILY_MEMBER_CHILD.AsGuid();
```

Comment:
Move to line 1389, closer to where you actually use it.

Reply:
Moved to global.